tests/rule_based_profiler/parameter_builder/test_parameter_computations.py

```diff
@@ -29,27 +32,136 @@ def _generate_distribution_samples(size: Optional[int] = 36) -> pd.DataFrame:
+            <= actual_false_positive_rates[column]
+            <= false_positive_rate + 0.01
+        )
+
+
+def test_compare_bootstrap_point_estimate_efficacy_with_bias_corrected():
+    experiment_repetitions = 20
+    # This works for sample size 1000 (used in other efficacy tests), but is very slow.
+    # Regardless, bootstrap is typically used for smaller sample sizes and should be tested as such.
+    # sample_size = 1000
+    sample_size = 200
+    false_positive_rate: np.float64 = np.float64(0.01)
+    lower_quantile_point_estimate: np.float64
+    upper_quantile_point_estimate: np.float64
+    lower_quantile_bias_corrected_point_estimate: np.float64
+    upper_quantile_bias_corrected_point_estimate: np.float64
+    actual_biased_false_positive_rates: Dict[str, Union[float, np.float64]] = {}
+    actual_bias_corrected_false_positive_rates: Dict[str, Union[float, np.float64]] = {}
+    improvement: list = []
+    for experiment in range(experiment_repetitions):
+        df: pd.DataFrame = _generate_distribution_samples(size=sample_size)
+        columns: pd.Index = df.columns
+        column: str
+        for column in columns:
+            (
+                lower_quantile_point_estimate,
+                upper_quantile_point_estimate,
+            ) = _compute_bootstrap_quantiles_point_estimate_legacy(
+                metric_values=df[column],
+                false_positive_rate=false_positive_rate,
+                n_resamples=DEFAULT_BOOTSTRAP_NUM_RESAMPLES,
+            )
+            actual_biased_false_positive_rates[column] = (
+                1.0
+                - np.sum(
+                    df[column].between(
+                        lower_quantile_point_estimate,
+                        upper_quantile_point_estimate,
+                    )
+                )
+                / df.shape[0]
+            )
+
+            (
+                lower_quantile_bias_corrected_point_estimate,
+                upper_quantile_bias_corrected_point_estimate,
+            ) = compute_bootstrap_quantiles_bias_corrected_point_estimate(
+                metric_values=df[column],
+                false_positive_rate=false_positive_rate,
+                n_resamples=DEFAULT_BOOTSTRAP_NUM_RESAMPLES,
+            )
+            actual_bias_corrected_false_positive_rates[column] = (
+                1.0
+                - np.sum(
+                    df[column].between(
+                        lower_quantile_bias_corrected_point_estimate,
+                        upper_quantile_bias_corrected_point_estimate,
+                    )
+                )
+                / df.shape[0]
+            )
+
+            improvement.append(
+                actual_biased_false_positive_rates[column]
+                >= actual_bias_corrected_false_positive_rates[column]
+            )
+
+    # bias correction should consistently improve performance at least 90% of the time
+    assert sum(improvement) / len(improvement) >= 0.90
```

Comment:
@NathanFarmer Very interesting result.  And we do not get an improvement by using the arithmetic means of the quantiles statistics based confidence intervals?  I am trying to find what I did, which also showed an improvement over the "biased" estimate.  Have we compared the said confidence intervals based approach with the implemented bias-correction based technique (while letting the bootstraps for the confidence intervals based approach be computed as part of `SciPy.bootstra()` method call?  Thank you.

Reply:
This is outdated. See new performance measurement methods.